### PR TITLE
Add object permission checker for caching permission lookup results

### PR DESCRIPTION
**Pull request by chrisvanrun**
Closes: #3388

Adds an object permission checker to the CIVSetList to more efficiently look up per-object permissions.

In addition:
 - It also adds a minor abstraction for the permission codenames
 - Pulls the "Title" column definition into the generic CIVSetList, this was initially combined with the new permission checker but I scraped this because of too much added complexity. Left the abstraction though.

### CODE REVIEW

File: app/grandchallenge/components/models.py [hunk 1]
```diff
@@ -2004,6 +2004,30 @@ def __content_str(self):
...
+
+    def save(self, *args, **kwargs):
+        adding = self._state.adding
+        super().save(*args, **kwargs)
+
+        if adding:
+            self.assign_permissions()
+
+    def assign_permissions(self):
+        pass
```
[the remark this answers is not in the document]
Comment (amickan) [reply app/grandchallenge/components/models.py:2028]: Yes, this should raise a NotImplementedError

File: app/grandchallenge/reader_studies/views.py [hunk 2]
```diff
@@ -384,22 +384,19 @@ class ReaderStudyStatistics(
...
-    ]
+
+    @property
+    def columns(self):
+        columns = CivSetListView.columns.copy()
+        columns.insert(
+            4,
+            Column(title="Order", sort_field="order"),
+        )
+        return columns
```
Comment (amickan) [inline app/grandchallenge/reader_studies/views.py:395]: Can you move this down so that it is below all directly defined attributes?

**Review by amickan — DISMISSED** [review]
LGTM

File: app/grandchallenge/components/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvanrun) [reply app/grandchallenge/components/models.py:2028]: Just my thoughts: this seems like a poor-man (but simple) implementation of ABC. However, it seems to follow the rest of the code so ImplementationError it is!

**Review by amickan — APPROVED** [review]

**Pull request merged**